api/account/api.go

```diff
@@ -0,0 +1,92 @@
+package account_api
+
+import (
+	"errors"
+	"regexp"
+
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+// CredentialRepresentation struct
+type CredentialRepresentation struct {
+	Id             *string `json:"id,omitempty"`
+	Type           *string `json:"type,omitempty"`
+	UserLabel      *string `json:"userLabel,omitempty"`
+	CreatedDate    *int64  `json:"createdDate,omitempty"`
+	CredentialData *string `json:"credentialData,omitempty"`
+	Value          *string `json:"value,omitempty"`
+	Temporary      *bool   `json:"temporary,omitempty"`
+}
+
+// UpdatePasswordBody is the definition of the expected body content of UpdatePassword method
+type UpdatePasswordBody struct {
+	CurrentPassword string `json:"currentPassword"`
+	NewPassword     string `json:"newPassword"`
+	ConfirmPassword string `json:"confirmPassword"`
+}
+
+type LabelBody struct {
+	Label string `json:"label,omitempty"`
+}
+
+// ConvertCredential creates an API credential from a KC credential
+func ConvertCredential(credKc *kc.CredentialRepresentation) CredentialRepresentation {
+	var cred CredentialRepresentation
+	cred.Id = credKc.Id
+	cred.Type = credKc.Type
+	cred.UserLabel = credKc.UserLabel
+	cred.CreatedDate = credKc.CreatedDate
+	cred.CredentialData = credKc.CredentialData
+	cred.Temporary = credKc.Temporary
```

Comment:
how about cred.Value?

Reply:
Value will be either used to transmit an info to Keycloak or to store a legacy secret. Thus we never want to return the value to final user.
I will add a comment about it as it not obvious at all.

Reply:
Value will be either used to transmit an info to Keycloak or to store a legacy secret. Thus we never want to return the value to final user.
I will add a comment about it as it not obvious at all.